pkg/prometheus/resource_selector.go

```diff
@@ -172,6 +173,7 @@ func (rs *ResourceSelector) SelectServiceMonitors(ctx context.Context, listFn Li
 				"namespace", objMeta.GetNamespace(),
 				"prometheus", objMeta.GetName(),
 			)
+			rs.metrics.Recorder.Eventf(sm, v1.EventTypeWarning, "InvalidConfiguration", "ServiceMonitor %q/%q was rejected due to invalid configuration: %v", sm.GetNamespace(), sm.GetName(), err)
```

Comment:
(nit) double quotes aren't needed IMHO.
```suggestion
			rs.metrics.Recorder.Eventf(sm, v1.EventTypeWarning, "InvalidConfiguration", "ServiceMonitor %s/%s was rejected due to invalid configuration: %v", sm.GetNamespace(), sm.GetName(), err)
```